pkg/util/circuit/options.go

```diff
@@ -41,7 +41,21 @@ type Options struct {
 	// not seeing any usage.
 	AsyncProbe func(report func(error), done func())
 
+	// SignalInterceptor gets to see and change the return value of the Signal
+	// method. This is useful for testing, as it allows "pretending" that the
+	// breaker is tripped without having to replace the probe method and
+	// coordinating with a running probe.
+	SignalInterceptor func(Signal) Signal
```

Comment:
Is this field stale? We have another one which is accessed via test only methods. Can't see any uses for this. If we use it in future then we should have some comment explaining what is the difference.

Reply:
This was dead code - thanks for spotting. There is an unexported copy of this below in the struct, which is what is actually being used.